### PR TITLE
Sam submit whitelist checking.

### PR DESCRIPTION
**Pull request by kshakir**
Switched the SamDenialResponse from "I don't know you" to "I know who your are and do not like you".

### CODE REVIEW

**Comment by kshakir** [comment]
**DO NOT MERGE** before https://github.com/broadinstitute/sam/pull/113 and related changes documented in the team drive.

File: CromIAM/src/main/scala/cromiam/sam/SamClient.scala [hunk 1]
```diff
@@ -24,11 +27,24 @@ class SamClient(scheme: String, interface: String, port: Int, log: LoggingAdapte
...
+      headers = List[HttpHeader](user.authorization)
+    )
 
+    for {
+      response <- Http().singleRequest(request)
+      whitelisted <- response.status match {
+        case StatusCodes.OK => Unmarshal(response.entity).to[String].map(_.toBoolean)
+        case _ => Future.successful(false)
+      }
+      _ = if (!whitelisted) log.error("Submit Access Denied for user {}", user.userId)
```
Comment (cjllanwarne) [inline CromIAM/src/main/scala/cromiam/sam/SamClient.scala]: ToL: `user.userId` is information that they gave us, right? Not that we've dug up from SAM for them?

File: CromIAM/src/main/scala/cromiam/sam/SamClient.scala [hunk 2]
```diff
@@ -98,15 +114,16 @@ class SamClient(scheme: String, interface: String, port: Int, log: LoggingAdapte
...
   private lazy val samBaseCollectionUri = s"$samBaseResourceUri/workflow-collection"
+  private lazy val samSubmitWhitelistUri = s"$samBaseResourceUri/caas/submit/action/get_whitelist"
 
 }
 
 object SamClient {
   case object SamDenialException extends Exception("Access Denied")
-  val SamDenialResponse = HttpResponse(status = StatusCodes.Unauthorized, entity = SamDenialException.getMessage)
+
+  val SamDenialResponse = HttpResponse(status = StatusCodes.Forbidden, entity = SamDenialException.getMessage)
```
Comment (cjllanwarne) [inline CromIAM/src/main/scala/cromiam/sam/SamClient.scala]: ToL: Oops, that comment above should probably have been here: we should make sure we don't leak anything from SAM in the exception message that the bad user didn't already tell us.

File: CromIAM/src/main/scala/cromiam/sam/SamClient.scala (hunk 1 above)
Comment (geoffjentry) [reply CromIAM/src/main/scala/cromiam/sam/SamClient.scala]: Look at how it is being called

**Pull request merged**